### DISCUSSION TITLE
TODO Triage List

### DISCUSSION THREAD

**Discussion by brianretford** [plaidml/plaidml#1821] [General]
Uncategorized:
- Replace EasyLogging w/ LLVM logging
- Migrate CI to GitHub Actions
- Create models repo / directory (deprecate IRv10 models repo)
- Look for replacements for Half
- Windows support
  - OpenMP doesn't support static linking on windows
- Support Tensorflow
   - Create plan - [TODO link to RFC]
     - Consider / investigate IREE strategy (may not cover MLIR TF bits - figure out how to get HLO in a sane way)
     - Consider GH action for building / packaging TF  
- Support blocked layouts
  - Support for hoisting constant/weight foldings into `init`
- Migration
  - Find new home for `drivers.vpu.devops` in 1source
- Fix SplitMain/Hoisting to avoid pack/unpack hack

P0:
- X86 Perf related
  - Implement Schedules (for loops) - per design [TODO link to RFC]
    - Attribute design
    - Adapt tiling 
    - Adapt fusion
  - Create x86 perf test suite
    - Add baseline (alex h hand code)
    - Add MLIR hand schedule
    - Add auto schedule test
  - Support manual tweaking of schedules
    - Add schedule override with support for Keras, OpenVINO, etc
    - Add 'args' support for pmlc-jit so that weights/acts can be patched in and evaluated
  - Improve cost models:
    - create more detailed plan [TODO link]
    - add support for multi cost models
    - table driven
    - a* search
    - heuristic

P1:

**Comment by deepdad** [comment]
Hi Brian, I just tried PlaidML. It's nice to see I can finally do something useful with my GPU on a Mac. How is development going? I see that the default installation uses deprecated keras stuff, pip install tensorflow (2.5) even messes up the plaidml installation. Then, for PyTorch, it looks like ONNX is used, which doesn't really support AMD/Mac, is it?
So I wonder how come this old keras does work?

**Comment by brianretford** [comment]
Hey,

There’s sort of been a bit of a development freeze for the past 3ish years. That’s not entirely accurate – we’ve been migrating to MLIR. I expect us to release a new (much newer) version in the next two months – that said, no idea how well various GPUs will work. There will be a solid foundation to work on though.

From: deepdad ***@***.***>
Date: Friday, August 6, 2021 at 1:50 AM
To: plaidml/plaidml ***@***.***>
Cc: Retford, Brian ***@***.***>, Author ***@***.***>
Subject: Re: [plaidml/plaidml] TODO Triage List (#1821)

Hi Brian, I just tried PlaidML. It's nice to see I can finally do something useful with my GPU on a Mac. How is development going? I see that the default installation uses deprecated keras stuff, pip install tensorflow (2.5) even messes up the plaidml installation. Then, for PyTorch, it looks like ONNX is used, which doesn't really support AMD/Mac, is it?
So I wonder how come this old keras does work?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/plaidml/plaidml/discussions/1821#discussioncomment-1138267>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEM5T2MPVICWKXW4PIK4UFTT3OO27ANCNFSM475NGZMQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.
